### PR TITLE
EVG-13787 add mana permissions for triggering git tag versions

### PR DESCRIPTION
**Pull request by ablack12**
This project was completed before we added granular options for auth, which is why users had to specify github usernames and teams directly on the project page. Now we can add this to mana as a regular entitlement (but we should still support the existing workflow). 

I added some changes to a different branch: https://github.com/evergreen-ci/evergreen/compare/main...ablack12:EVG-13787-part2 for after MANA adds the new setting.

### CODE REVIEW

File: model/project_ref.go [hunk 1]
```diff
@@ -1475,11 +1475,31 @@ func (p *ProjectRef) UpdateAdminRoles(toAdd, toRemove []string) error {
...
+	if utility.StringSliceContains(p.GitTagAuthorizedUsers, githubUser) {
 		return true
 	}
-	return thirdparty.IsUserInGithubTeam(ctx, p.GitTagAuthorizedTeams, p.Owner, user, token)
+	// check if user has permissions with mana before asking github about the teams
+	u, err := user.FindByGithubName(githubUser)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "error checking if user is authorized for git tag",
+			"source":  "github hook",
```
Comment (malikchaya2) [inline model/project_ref.go:1487]: This seems to be a db check, and we don't seem to be accessing github until line 1502. Why do we want the source to be "github hook"?

Comment (ablack12) [reply model/project_ref.go:1487]: I'm using "github hook" to keep this error message linked to the logs that come from the Github webhook (which is the only time we use this function)

Comment (malikchaya2) [reply model/project_ref.go:1487]: makes sense, thanks!

**Review by malikchaya2 — APPROVED** [review]

**Review by john-m-liu — APPROVED** [review]
could you also add EVG-14525 in the pr/commit title so that the jira bot picks this up? well assuming it doesn't mess up like it's been doing lately

**Comment by ablack12** [comment]
I don't want to clutter the commit title but I'm going to close that ticket as already done and link this ticket.

**Pull request merged**